staging/src/k8s.io/client-go/tools/watch/informerwatcher.go

```diff
@@ -137,13 +152,31 @@ func NewIndexerInformerWatcher(lw cache.ListerWatcher, objType runtime.Object) (
 		},
 	}, cache.Indexers{})
 
+	// This will get stopped, but without waiting for it.
 	go e.run()
 
+	logger := klog.FromContext(ctx)
+	if ctx.Done() != nil {
+		// Optionally map cancellation to w.Stop. The informer below
+		// only waits for that.
+		go func() {
+			select {
+			case <-ctx.Done():
+			case <-w.StopChan():
+			}
+			w.Stop()
```

Comment:
why do we need that after `<-w.StopChan()`?

Reply:
That was the "Optionally map cancellation..." comment above. It's only relevant if the context is done. While it doesn't hurt to always call it, it may be more obvious why it gets called when its in the `case <-ctx.Done()` - moved and updated the comment.

Reply:
Removed entirely as part of the NewIndexerInformerWatcherWithContext -> NewIndexerInformerWatcherWithLogger simplification (to be confirmed, see https://github.com/kubernetes/kubernetes/pull/129341#discussion_r1935556454).